contracts/colonyNetwork/ColonyNetworkSkills.sol

```diff
@@ -71,91 +396,130 @@ contract ColonyNetworkSkills is ColonyNetworkStorage, Multicall {
+        uint256 _newSkillId = skill.parents[i];
+        uint256 _newParentSkillNumber = _parentSkillNumber - 2 ** i;
+        return ascendSkillTree(_newSkillId, _newParentSkillNumber);
+      }
+    }
   }
 
-  function appendReputationUpdateLog(
+  function appendReputationUpdateLogInternal(
     address _user,
-    int _amount,
-    uint _skillId
-  ) public stoppable calledByColony skillExists(_skillId) {
-    if (_amount == 0 || _user == address(0x0)) {
-      // We short-circut amount=0 as it has no effect to save gas, and we ignore Address Zero because it will
-      // mess up the tracking of the total amount of reputation in a colony, as that's the key that it's
-      // stored under in the patricia/merkle tree. Colonies can still pay tokens out to it if they want,
-      // it just won't earn reputation.
-      return;
-    }
-
+    int256 _amount,
+    uint256 _skillId,
+    address _colony
+  ) internal {
     uint128 nParents = skills[_skillId].nParents;
     // We only update child skill reputation if the update is negative, otherwise just set nChildren to 0 to save gas
-    uint128 nChildren = _amount < 0 ? skills[_skillId].nChildren : 0;
+    uint128 nChildren = (_amount < 0) ? skills[_skillId].nChildren : 0;
     IReputationMiningCycle(inactiveReputationMiningCycle).appendReputationUpdateLog(
       _user,
       _amount,
       _skillId,
-      msgSender(),
+      _colony,
       nParents,
       nChildren
     );
   }
 
-  // View
+  function bridgeReputationUpdateLog(address _user, int256 _amount, uint256 _skillId) internal {
+    // TODO: Maybe force to be set on deployment?
+    require(colonyBridgeAddress != address(0x0), "colony-network-foreign-bridge-not-set");
+    address colonyAddress = msgSender();
+    reputationUpdateCount[block.chainid][colonyAddress] += 1;
+    // Build the transaction we're going to send to the bridge
+    bytes memory payload = abi.encodeWithSignature(
+      "addReputationUpdateLogFromBridge(address,address,int256,uint256,uint256)",
+      colonyAddress,
+      _user,
+      _amount,
+      _skillId,
+      reputationUpdateCount[block.chainid][colonyAddress]
+    );
 
-  function getParentSkillId(uint _skillId, uint _parentSkillIndex) public view returns (uint256) {
-    return ascendSkillTree(_skillId, _parentSkillIndex + 1);
-  }
+    bool success = callThroughBridgeWithGuards(payload);
 
-  function getChildSkillId(uint _skillId, uint _childSkillIndex) public view returns (uint256) {
-    if (_childSkillIndex == UINT256_MAX) {
-      return _skillId;
-    } else {
-      Skill storage skill = skills[_skillId];
-      require(
-        _childSkillIndex < skill.children.length,
-        "colony-network-out-of-range-child-skill-index"
+    if (success) {
+      emit ReputationUpdateSentToBridge(
+        colonyAddress,
+        reputationUpdateCount[block.chainid][colonyAddress]
       );
-      return skill.children[_childSkillIndex];
+      return;
```

Comment:
Why not do an `else` here?

Reply:
This is the same pattern as came up recently, namely that I don't want to forget that we introduced the `else` to implicitly mean 'and do nothing else in the if branch', and then introduce something after the `if` statement at a later date that breaks that assumption.